### ISSUE TITLE
Possible memory leak with unserialize

### ISSUE THREAD

**Issue by dkarlovi** [igbinary/igbinary#360]
Looking at [this graph](https://blackfire.io/profiles/86b8418e-6202-4932-9924-7cb6566f576a/graph) memory usage (select 355MB in top toolbar), it seems `igbinary_unserialize` is leaking memory when called a bunch of times. I'm not sure if Blackfire just misreports this (the memory is actually used by objects which unserialize spawned (?) or this might be an actual memory leak?

What would be the easiest way to tell?

**Comment by TysonAndre** [comment]
Also, I'd assume whatever memory usage is reported would be based on emalloc/memory_get_usage(), which tracks request-bound allocations (i.e. not persistent leaks)

igbinary's own tests would fail with `make test TESTS=-m` if there was an obvious request-bound leak, and igbinary doesn't use permanent allocations

**Comment by dkarlovi** [comment]
In this case it's CLI, but GC is running constantly and in theory it shouldn't keep any references around, in any obvious way anyway. The underlying system is quite thick so I'll check some other way.

Thank you very much for the confirmation, have a nice day!

**Comment by dkarlovi** [comment]
Coming back to this, I'm still seeing the same behaviour with PHP 8.0 and 8.1 on Alpine. Then it occured to me you've noted this would be seen by the test suite. Checking the CI scripts, it seems the tests are running on Ubuntu / Debian? They're both glibc based, while Alpine is musl based.

@TysonAndre are the tests being run on Alpine anywhere? Could this be an Alpine-specific regression?

**Comment by TysonAndre** [comment]
No, they aren't, I'd expect alpine and ubuntu to work the same way.
